src/swc.rs

```diff
@@ -369,6 +382,24 @@ fn infer_name_from_expr(mut expr: &ast::Expr) -> Option<ScopeName> {
     }
 }
 
+fn computed_prop_name_to_component(prop: &ast::ComputedPropName) -> Option<(NameComponent, bool)> {
+    if let Some(lit) = prop.expr.as_lit() {
+        if let ast::Lit::Str(prop) = lit {
+            return Some((NameComponent::interp(prop.value.to_string()), true));
+        }
+
+        if let ast::Lit::Num(prop) = lit {
+            return Some((NameComponent::interp(format!("[{}]", prop.value)), false));
+        }
+    }
+
+    if let Some(ident) = prop.expr.as_ident() {
+        return Some((NameComponent::interp(format!("[{}]", ident.sym)), false));
```

Comment:
the identifier itself could be a valid token that we can remap, turning `[a]` into `[name]`. so we could use:
`interp("["), ident(ident), interp("]")`